core/rio/hdt/src/main/java/org/eclipse/rdf4j/rio/hdt/part/HDTTriplesSectionFactory.java

```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.rio.hdt.part;
+
+import java.io.IOException;
+
+/**
+ * HDT Array factory.
+ * 
+ * @author Bart Hanssens
+ */
+public class HDTTriplesSectionFactory {
+
+	public static HDTTriplesSection parse(String str) throws IOException {
+		if (!str.equals(new String(HDTTriples.FORMAT_BITMAP))) {
```

Comment:
Not clear why this is being round-tripped. There should be another variable with the String version to directly compare to.